src/Altinn.Platform/Altinn.Platform.Register/Register/Program.cs

```diff
@@ -1,164 +1,286 @@
+    config.SetBasePath(basePath);
+    config.AddJsonFile(basePath + "altinn-appsettings/altinn-dbsettings-secret.json", optional: true, reloadOnChange: true);
+    if (basePath == "/")
+    {
+        config.AddJsonFile(basePath + "app/appsettings.json", optional: false, reloadOnChange: true);
+    }
+    else
     {
-        private static ILogger _logger;
+        config.AddJsonFile(Directory.GetCurrentDirectory() + "/appsettings.json", optional: false, reloadOnChange: true);
+    }
 
-        /// <summary>
-        /// Default protected constructor
-        /// </summary>
-        protected Program()
-        {
-        }
+    config.AddEnvironmentVariables();
 
-        /// <summary>
-        /// The main method
-        /// </summary>
-        /// <param name="args">The Arguments</param>
-        public static void Main(string[] args)
-        {
-            ConfigureSetupLogging();
-            CreateWebHostBuilder(args).Build().Run();
-        }
+    ConnectToKeyVaultAndSetApplicationInsights(config);
 
-        /// <summary>
-        /// Configure logging for setting up application. Temporary
-        /// </summary>
-        public static void ConfigureSetupLogging()
+    config.AddCommandLine(args);
+}
+
+void ConnectToKeyVaultAndSetApplicationInsights(ConfigurationManager config)
+{
+    KeyVaultSettings keyVaultSettings = new KeyVaultSettings();
+    config.GetSection("kvSetting").Bind(keyVaultSettings);
+    if (!string.IsNullOrEmpty(keyVaultSettings.ClientId) &&
+        !string.IsNullOrEmpty(keyVaultSettings.TenantId) &&
+        !string.IsNullOrEmpty(keyVaultSettings.ClientSecret) &&
+        !string.IsNullOrEmpty(keyVaultSettings.SecretUri))
+    {
+        logger.LogInformation("Program // Configure key vault client // App");
+
+        string connectionString = $"RunAs=App;AppId={keyVaultSettings.ClientId};" +
+                                  $"TenantId={keyVaultSettings.TenantId};" +
+                                  $"AppKey={keyVaultSettings.ClientSecret}";
+        AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider(connectionString);
+        KeyVaultClient keyVaultClient = new KeyVaultClient(
+            new KeyVaultClient.AuthenticationCallback(
+                azureServiceTokenProvider.KeyVaultTokenCallback));
+        config.AddAzureKeyVault(
+            keyVaultSettings.SecretUri, keyVaultClient, new DefaultKeyVaultSecretManager());
+        try
         {
-            // Setup logging for the web host creation
-            var logFactory = LoggerFactory.Create(builder =>
-            {
-                builder
-                    .AddFilter("Altinn.Platform.Register.Program", LogLevel.Debug)
-                    .AddConsole();
-            });
-
-            _logger = logFactory.CreateLogger<Program>();
-        }
+            SecretBundle secretBundle = keyVaultClient
+                .GetSecretAsync(keyVaultSettings.SecretUri, vaultApplicationInsightsKey).Result;
```

Comment:
This call should use await. This means the method and callers must return async Task instead, but that should be fine.